### PR TITLE
Remove localDisclosures from BlindingInfo entirely

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Addresses https://github.com/digital-asset/daml/pull/6507#discussion_r446113575

Builds on top of @hurryabit's [clean up](https://github.com/digital-asset/daml/pull/6507) and in particular @remyhaemmerle-da's work to get rid of relative contract identifiers.

### CODE REVIEW

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingCoder.scala [hunk 1]
```diff
@@ -25,13 +25,6 @@ object BlindingCoder {
           parties <- toPartySet(n.getPartiesList)
         } yield ni -> parties)
 
-    val implicitLocal =
-      p.getLocalImplicitDisclosureList.asScala.map(n =>
```
Comment (hurryabit) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingCoder.scala:29]: This field still exists on the protobuf message. We're now just ignoring it. Is that the right thing to do or should we rather assert that it is empty?

Comment (stefanobaghino-da) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingCoder.scala:29]: I'm not entirely sure. Could an assert end up causing issues with backwards compatibility?

Comment (hurryabit) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingCoder.scala:29]: If it caused issues with backward compatibility, then ignoring the field would be the wrong thing, wouldn't it? @gerolf-da @remyhaemmerle-da do you know more?

Comment (gerolf-da) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingCoder.scala:29]: I don't think we use the `BlindingInfo` protobuf message anywhere. Whenever `com.daml.lf.transaction.BlindingInfo` is needed, it is computed from the `Transaction` itself by using one of the `com.daml.lf.engine.Blinding` functions.
@remyhaemmerle-da can you confirm that? Might be worth checking with other ledger integrations to be sure though.

Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingCoder.scala:29]: I was not really aware of this proto encoding.
It seems to be completly dead code with the `BlindingCoder` class and a test.

Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/BlindingCoder.scala:29]: I push a commit where I drop this useless file.

**Pull request merged**